src/Helper/Cron/Cancellation.php

```diff
@@ -0,0 +1,246 @@
+            } else {
+                // Received error from shopgate server
+                $this->messageManager->addErrorMessage(
+                    __(
+                        '[SHOPGATE] An error occured while trying to cancel the order at Shopgate.<br />Please contact Shopgate support.'
+                    )
+                );
+                $this->messageManager->addErrorMessage("Error: {$e->getCode()} - {$e->getMessage()}");
+                $this->logger->error(
+                    "(#{$orderNumber})  SMA-Error on cancel order! Message: {$e->getCode()} - {$e->getMessage()}"
+                );
+            }
+        } catch (\Exception $e) {
+            $this->messageManager->addErrorMessage(
+                __('[SHOPGATE] An unknown error occured!<br />Please contact Shopgate support.')
+            );
+            $this->logger->error(
+                "(#{$orderNumber})  SMA-Error on cancel order! Message: {$e->getCode()} - {$e->getMessage()}"
+            );
+        }
+    }
+
+    /**
+     * @param ShopgateOrderModel $shopgateOrderModel
+     * @param MagentoOrder       $magentoOrder
+     */
+    protected function finalizeCancellation($shopgateOrderModel, $magentoOrder)
+    {
+        $this->messageManager->addSuccessMessage(__('[SHOPGATE] Order successfully cancelled at Shopgate.'));
+
+        $shopgateOrderModel->setIsCancellationSentToShopgate(true);
+        $shopgateOrderModel->save();
+
+        if (!$shopgateOrderModel->getIsSentToShopgate()
+            && !$this->shippingHelper->completeShipping(
+                $shopgateOrderModel,
+                $magentoOrder
+            )
+        ) {
+            $this->logger->error(
+                "! (#{$shopgateOrderModel->getShopgateOrderNumber()})  not sent to shopgate and shipping not complete"
+            );
+        }
+    }
+
+    /**
+     * @param array $cancelledItems
+     *
+     * @return int
+     */
+    protected function getCancelledItemCount($cancelledItems)
+    {
+        $qtyCancelled = 0;
+
+        foreach ($cancelledItems as $cancelledItem) {
+            $qtyCancelled += $cancelledItem['quantity'];
+        }
+
+        return $qtyCancelled;
+    }
+
+    /**
+     * @param Magento\Sales\Model\ResourceModel\Order\Item\Collection $orderItems
+     * @param \ShopgateOrder                                          $shopgateOrder
+     *
+     * @return array
+     */
+    protected function getItemsForCancellation($orderItems, $shopgateOrder)
+    {
+        $cancellationItems = [];
+
+        /**  @var $orderItem \Magento\Sales\Model\Order\Item */
+        foreach ($orderItems as $orderItem) {
+            $rdItem   = $this->cronHelper->findItemByProductId(
+                $shopgateOrder->getItems(),
+                $orderItem->getData('product_id')
+            );
+            $mainItem = empty($orderItem->getParentItem()) || $orderItem->getParentItem()->getProductType() != Configurable::TYPE_CODE
```

Comment:
`getParentItem()` may be null and throw when `getProductType()` is called

Reply:
fixed